### PR TITLE
Add Rehearsal / Choreography for 1 Button

### PR DESCRIPTION
**Pull request by fdarosa2663**

### CODE REVIEW

File: props/saber_fett263_buttons.h [hunk 1]
```diff
@@ -4653,7 +4723,11 @@ SaberFett263Buttons() : PropBase() {}
 	  break;
 	}
         if (!SaberBase::Lockup()) {
-          SaberBase::SetLockup(SaberBase::LOCKUP_NORMAL);
+          if (fusor.angle1() < - M_PI / 4) {
+            SaberBase::SetLockup(SaberBase::LOCKUP_DRAG);
```
Comment (profezzorn) [inline props/saber_fett263_buttons.h:4727]: This doesn't seem to have anything to do with choreography.

File: props/saber_fett263_buttons.h [hunk 2]
```diff
@@ -4285,18 +4317,48 @@ SaberFett263Buttons() : PropBase() {}
...
+#ifdef FETT263_SAVE_CHOREOGRAPHY
+      // Rehearsal Mode
+      case EVENTID(BUTTON_POWER, EVENT_THIRD_HELD_LONG, MODE_OFF):
+        if (rehearse_) {
+          sound_library_.SayCancel();
+          rehearse_ = false;
+          return true;
+        }
+        // Check for existing rehearsal and prompt to overwrite or keep via menu
+        if (saved_choreography.clash_rec[0].stance == SavedRehearsal::STANCE_CLASH || saved_choreography.clash_rec[0].stance == SavedRehearsal::STANCE_LOCKUP) {
```
Comment (profezzorn) [inline props/saber_fett263_buttons.h:4342]: I assume this just mean "if choreography is initialized (by recording or loading)".
A helper function in saved_choreography called "isInitialized" or similar would make that much more clear.

File: props/saber_fett263_buttons.h (hunk 1 above)
Comment (fdarosa2663) [reply props/saber_fett263_buttons.h]: Will submit separately

File: props/saber_fett263_buttons.h (hunk 2 above)
Comment (fdarosa2663) [reply props/saber_fett263_buttons.h]: updated

**Comment by fdarosa2663** [comment]
fixed a few small things during testing for 1 button to improve use

**Pull request merged**